src/cascadia/TerminalApp/App.cpp

```diff
@@ -827,7 +828,7 @@ namespace winrt::TerminalApp::implementation
             const auto profiles = _settings->GetProfiles();
 
             // If we don't have that many profiles, then do nothing.
-            if (realIndex >= profiles.size())
+            if (realIndex >= gsl::narrow_cast<decltype(realIndex)>(profiles.size()))

```

Comment:
If the type is changed for any reason, one would not need to also change that line.
That said, this was a small mistake as I should have used `gsl::narrow` instead. It will raise an exception if the cast would change the value in any way. (see [gsl_util](https://github.com/Microsoft/GSL/blob/master/include/gsl/gsl_util)).
I'll wait for you feedback to make the change.

Reply:
This seems correct to me. I love decltype.